keps/420-partial-admission/README.md

```diff
@@ -0,0 +1,148 @@
+In case a partial fit is chosen, the jobframework reconciler should provide the admitted pod counts to the parent job before unsuspending it, in a similar fashion as the node selectors are provided. In case the job gets suspended, the original pod counts should be restored in order to allow a potential future admission with its original pod counts.
+
+### User Stories
+
+Kueue issue [420](https://github.com/kubernetes-sigs/kueue/issues/420) provides details on the initial feature details and its applicability for `batch/Job`.
+
+### Risks and Mitigations
+
+TBD: It should be decided if the partial admission should be taken into account before attempting preemption. Since preemption has its own penalties (time to evict the active workloads, potential waste of the resources used by the preempted workloads) I vote to take it into account before.
+
+## Design Details
+
+### Workload API
+
+```go
+type PodSet struct {
+    // .......
+
+    // count is the number of pods for the spec.
+    // +kubebuilder:validation:Minimum=1
+    Count int32 `json:"count"`
+
+    // minimumCount is the minimum number of pods for the spec acceptable
+    // in case of partial admission.
+    //
+    // If not provided, partial admission for the current PodSet is not
+    // enabled.
+    // +optional
+    MinimumCount *int32 `json:"minimumCount,omitempty"`
+}
+
+```
+
+### Scheduler / Flavorassignment
+
+I case the current workload dose not in the current available quota and any of its PodSets allow partial admission and the cont did not reach the `MinimumCount`, decrease the cont in one PodSet and try again. The accepted number of pods in each PodSet are published in `workload.Status.Admission.PodSetAssignments[*].ResourceUsage.Count`
+
+### Jobframework
+
+```diff
+type GenericJob interface {
+    // ...
+
+	
+-    // RunWithNodeAffinity will inject the node affinity extracting from workload to job and unsuspend the job.
++    // RunWithPodSetsInfo will inject the node affinity and podsSet counts extracted from workload to the job and unsuspend the job.
+-    RunWithNodeAffinity(nodeSelectors []PodSetNodeSelector)
++    RunWithPodSetsInfo(nodeSelectors []PodSetNodeSelector, podSetCounts []int32)
+-    // RestoreNodeAffinity will restore the original node affinity of job.
++    // RestorePodSetsInfo will restore the original node affinity of job.
+-    RestoreNodeAffinity(nodeSelectors []PodSetNodeSelector)
++    RestorePodSetsInfo(nodeSelectors []PodSetNodeSelector, podSetCounts []int32)
+
+    // ...
+}
+
+```
+
+### batch/Job controller
+
+Besides adapting `RunWithPodSetsInfo` and `RestorePodSetsInfo` it should also:
+
+- rework `PodSets()` to populate `MinimumCount` if the job is marked to support partial admission.
+  * jobs supporting partial admission should have a dedicated annotation. eg. `kueue.x-k8s.io/accept-partial-admission`, indicating the minimum `parallelism` and other aspect that should be taken into consideration like elastic index jobs.
```

Comment:
can you provide an example of this? Are you saying that the `accept-partial-admission` should contain a number?

Reply:
changed to `kueue.x-k8s.io/job-min-parallelism` , should make more sense.